service/history/ndc/workflow_resetter.go

```diff
@@ -762,6 +767,7 @@ func (r *workflowResetterImpl) reapplyEventsFromBranch(
 	nextEventID int64,
 	branchToken []byte,
 	resetReapplyExcludeTypes map[enumspb.ResetReapplyExcludeType]struct{},
+	allowResetWithPendingChildren bool,
 	childrenInitializedAfterReset map[string]*persistencespb.ResetChildInfo,
```

Comment:
Thought of sending a nil value of `childrenInitializedAfterReset` (when allowResetWithPendingChildren is false) instead of passing the flag all the way here. But check for nil map and then populating it is an antipattern. So just decided to pass the flag explicitly.